crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -1227,70 +1227,202 @@ pub struct Segment2d {
 impl Primitive2d for Segment2d {}
 
 impl Segment2d {
-    /// Create a new `Segment2d` from its endpoints
+    /// Create a new `Segment2d` from its endpoints.
     #[inline(always)]
     pub const fn new(point1: Vec2, point2: Vec2) -> Self {
         Self {
             vertices: [point1, point2],
         }
     }
 
-    /// Create a new `Segment2d` from its endpoints and compute its geometric center
-    ///
-    /// # Panics
-    ///
-    /// Panics if `point1 == point2`
+    /// Create a new `Segment2d` from its endpoints and compute its geometric center.
     #[inline(always)]
     #[deprecated(since = "0.16.0", note = "Use the `new` constructor instead")]
     pub fn from_points(point1: Vec2, point2: Vec2) -> (Self, Vec2) {
         (Self::new(point1, point2), (point1 + point2) / 2.)
     }
 
-    /// Create a new `Segment2d` at the origin from a `direction` and `length`
+    /// Create a new `Segment2d` at the origin with the given direction and length.
```

Comment:
```suggestion
    /// Create a new `Segment2d` centered at the origin with the given direction and length.
```

"at the origin" feels a bit ambiguous - could mean "starting from the origin". Also considered "through the origin".

Reply:
"centered at the origin" seems best to me.